STMicroelectronics/STM32L4_L4+/app/az_ulib_dm/src/az_ulib_registry.c

```diff
@@ -0,0 +1,300 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+#include "_az_ulib_registry_interface.h"
+#include "az_ulib_ipc_api.h"
+#include "az_ulib_ipc_interface.h"
+#include "az_ulib_registry_api.h"
+#include "az_ulib_result.h"
+#include "stm32l475_flash_driver.h"
+#include <azure/core/internal/az_precondition_internal.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+
+extern uint32_t __REGISTRYINFO;
+extern uint32_t __REGISTRY;
+
+static az_ulib_pal_os_lock registry_lock;
+
+static az_result result_from_hal_status(HAL_StatusTypeDef status)
+{
+  switch (status)
+  {
+    case HAL_OK:
+      return AZ_OK;
+    case HAL_ERROR:
+      return AZ_ERROR_ULIB_SYSTEM;
+    case HAL_BUSY:
+      return AZ_ERROR_ULIB_BUSY;
+    case HAL_TIMEOUT:
+      return AZ_ERROR_ULIB_SYSTEM; // should be AZ_ERROR_ULIB_TIME_OUT
+    default:
+      return AZ_ERROR_ULIB_SYSTEM;
+  }
+}
+
+static az_result set_registry_node_ready_flag(uint8_t* address)
+{
+  AZ_ULIB_TRY
+  {
+    // if ((hal_status = internal_flash_write_doubleword((uint32_t)address, REGISTRY_READY)) !=
+    // HAL_OK)
+    // {
+    //   result = result_from_hal_status(hal_status);
+    //   AZ_ULIB_THROW_IF_AZ_ERROR(result);
+    // }
+    AZ_ULIB_THROW_IF_AZ_ERROR(
+        result_from_hal_status(internal_flash_write_doubleword((uint32_t)address, REGISTRY_READY)));
```

Comment:
>result_from_hal_status(internal_flash_write_doubleword((uint32_t)address, REGISTRY_READY))); [](http://example.com/codeflow?start=8&length=92)

if this is the only function you're calling you can just return this and let it be caught at the caller level.